### PR TITLE
Handle undefined/wrong-type error in service request

### PR DESCRIPTION
**Pull request by AlexKamaev**
This is a duplicate of https://github.com/DevExpress/testcafe-hammerhead/pull/2481/ with rebase, tests and some modifications

### CODE REVIEW

**Comment by AlexKamaev** [comment]
@testcafe-build-bot retest

**Comment by AlexKamaev** [comment]
it's ready for review. Tests are failed because of some problems with SauceLabs

**Comment by AlexKamaev** [comment]
@testcafe-build-bot retest

File: src/utils/logger.ts [hunk 1]
```diff
@@ -3,6 +3,13 @@ import RequestPipelineContext from '../request-pipeline/context';
 import { IncomingMessage, OutgoingHttpHeaders } from 'http';
 import { ServiceMessage } from '../typings/proxy';
 import RequestOptions from '../request-pipeline/request-options';
+import errToString from './err-to-string'
+
+function getIncorrectErrorTypeMessage (err: object) {
+    const errType = typeof err;
+
+    return `The "${errToString(err)}" error of the "${errType}" type was passed. Make sure that service message handlers throw errors correctly.`;
```
Comment (miherlosev) [inline src/utils/logger.ts:11]: I propose another error message.

```js
Incorrect "${errType}" error was passed to the error handler. Make sure that service message handlers throw errors correctly.
```

**Review by miherlosev — CHANGES_REQUESTED** [review]

File: src/utils/logger.ts (hunk 1 above)
Comment (AlexKamaev) [reply src/utils/logger.ts:11]: Don't you want to show the error text if it exists?
for example:

```JS
throw 'string error';
```
`The "string error" error of the "string" type was passed. `

Comment (miherlosev) [reply src/utils/logger.ts:11]: Ok. Let it be.

**Review by miherlosev — APPROVED** [review]

File: src/utils/logger.ts (hunk 1 above)
Comment (AlexKamaev) [reply src/utils/logger.ts:11]: @DIRECTcut check this message please:
`The "${errToString(err)}" error of the "${errType}" type was passed. Make sure that service message handlers throw errors correctly.`

**Comment by AlexKamaev** [comment]
@LavrovArtem review please

File: src/utils/logger.ts (hunk 1 above)
Comment (DIRECTcut) [reply src/utils/logger.ts:11]: It's unclear what `correctly` means. For example, does it mean:

`The "${errToString(err)}" error of the "${errType}" type was passed. Make sure that service message handlers throw errors of a correct type.`

Comment (AlexKamaev) [reply src/utils/logger.ts:11]: the error should be of Error type.
In JS you can make the following:
```JS
throw 1
```
but the correct usage is:
```JS
throw new Error('1');
```
What to do you think if I change it to
`Make sure that service message handlers throw errors of the Error type`

**Review by LavrovArtem — APPROVED** [review]

**Review by DIRECTcut — APPROVED** [review]

**Pull request merged**